pkg/loki/modules.go

```diff
@@ -450,6 +450,13 @@ func (t *Loki) initQuerier() (services.Service, error) {
 func (t *Loki) initIngester() (_ services.Service, err error) {
 	t.Cfg.Ingester.LifecyclerConfig.ListenPort = t.Cfg.Server.GRPCListenPort
 
+	if t.Cfg.Ingester.ShutdownMarkerPath == "" && t.Cfg.Common.PathPrefix != "" {
+		t.Cfg.Ingester.ShutdownMarkerPath = t.Cfg.Common.PathPrefix
+	}
+	if t.Cfg.Ingester.ShutdownMarkerPath == "" {
```

Comment:
By having this check we'll break all Loki installations that aren't specifying a `common_prefix`. Since this is a minor version we'd be giving a better experience to users by avoiding that.

Suggestion: Log a warning here and move this check to the `PrepareShutdown` endpoint. Panic'ing on `PrepareShutdown` is better imo because a user using it knows what to expect; won't be surprised by the new marker change.

Reply:
That is a very good suggestion. It has been implemented like that.